stan/math/opencl/kernel_generator/evaluate_into.hpp

```diff
@@ -0,0 +1,72 @@
+#ifndef STAN_MATH_OPENCL_KERNEL_GENERATOR_EVALUATE_INTO_HPP
+#define STAN_MATH_OPENCL_KERNEL_GENERATOR_EVALUATE_INTO_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/kernel_generator/operation.hpp>
+#include <stan/math/opencl/kernel_generator/as_operation.hpp>
+#include <stan/math/opencl/kernel_generator/is_valid_expression.hpp>
+#include <cl.hpp>
+#include <utility>
+#include <string>
+#include <set>
+
+namespace stan {
+namespace math {
+
+template <typename Derived, typename ReturnScalar>
+template <typename T_lhs>
+void operation<Derived, ReturnScalar>::evaluate_into(T_lhs&& lhs) const {
+  using enable = enable_if_all_valid_expressions<T_lhs>;
+  using cache = operation<Derived, ReturnScalar>::cache<T_lhs>;
+  auto lhs_expression = as_operation(std::forward<T_lhs>(lhs));
+
+  int n_rows = derived().rows();
+  int n_cols = derived().cols();
+  const char* function = "evaluate_into";
+  if (n_rows != dynamic) {
```

Comment:
I don't like having `this->` before static members.

Reply:
I have a big preference for it, especially since this function is defined outside the class and I have to track around looking for where `dynamic` was defined and whether something out of the class is being called. Easy to imagine a situation like

```cpp
// in one header
using dynamic = Eigen::Dynamic;

// another header
class foo {
  int dynamic{1};
  void eval();
};

// someone else writing another header
class goo : foo {
  Eigen::Matrix<double, dynamic, dynamic> A;
};
```

the last member init would be a lot more clear with

```cpp
Eigen::Matrix<double, this->dynamic, this->dynamic> A;
```